packages/core/lib/configAdapter.js

```diff
@@ -0,0 +1,103 @@
+function sanitizeGanacheOptions(ganacheOptions) {
+  const network_id = ganacheOptions.network_id;
+
+  // Use default network_id if "*" is defined in config
+  if (network_id === "*") {
+    return { ...ganacheOptions, network_id: Date.now() };
+  }
+
+  const parsedNetworkId = parseInt(network_id, 10);
+  if (isNaN(parsedNetworkId)) {
+    const error =
+      `The network id specified in the truffle config ` +
+      `(${network_id}) is not valid. Please properly configure the network id as an integer value.`;
+    throw new Error(error);
+  }
+
+  return { ...ganacheOptions, network_id: parsedNetworkId };
+}
+
+// This function returns the first defined argument value and throws if all arguments are undefined
+const getFirstDefinedValue = (...values) => {
```

Comment:
Btw, if you want to make this shorter, you can just do `const getFirstDefinedValue = (...values) => values.find(value => value !== undefined)`. :)  Well, that version won't throw an error if nothing's passed in, it'll return `undefined`, but I don't see why we need to throw an error in that case anyway.

Reply:
Harry's comment makes sense. Returning undefined may be the better way to proceed. Ganache already has sensible defaults which we could rely on.

Reply:
This makes sense to me too. There is probably no need of throwing an error and we can leave it undefined.